### DISCUSSION TITLE
TCP connection ignored on C/C++ Remoted Debugging

### DISCUSSION THREAD

**Discussion by mmoschetti** [eclipse-cdt/cdt#370] [Q&A]
Hello,
I have a C/C++ Remote Debugging configuration (Manual) with Debbugger set to use a TCP connection, but when I start it I receive this error:

"Failed to execute MI command:
-target-select remote /dev/ttyS0
Error message from debugger back end:
/dev/ttyS0: No such file or directory.
Failed to execute MI command:
-target-select remote /dev/ttyS0
Error message from debugger back end:
/dev/ttyS0: No such file or directory.
/dev/ttyS0: No such file or directory."

It was working correctly before updating Eclipse Photon (June 2018) with 2023.03.
Starting the new Eclipse with the same workspace all seems correctly configured.
If I open "Debug Configurations..." -> "Debugger" -> "Connection" I see that "Type" is properly configured to use "TCP". IP address and port are correctly configured but the configuration is ignored.
If I switch to "Serial" and I change the "Device" field to a different one the error reflects the change: "Failed to execute MI command:
-target-select remote /dev/ttyS1" (also when set to "TCP")

Can anyone help me?

**Comment by jld01** [comment]
@mmoschetti can you attach the `*.launch` file containing the problem launch configuration please? This may help in diagnosing this issue. The launch file may be located in your project, or it may be in the `.metadata/.plugins/org.eclipse.debug.core/.launches/` folder under your Eclipse workspace folder.

**Comment by mmoschetti** [comment]
Thanks for your reply.
I exported the configuration and attached it (renamed from .launch to .txt).
Using the "Automatic Remote Debugging Launcher" works with TCP without problems.

[Z105A-Release Configuration.txt](https://github.com/eclipse-cdt/cdt/files/11273028/Z105A-Release.Configuration.txt)

**Reply by jld01** [reply]
@mmoschetti your launch configuration looks correct and triggers a TCP connection request as expected with latest CDT under debug:

`-target-select remote 192.168.9.10:2345`

I suggest you try comparing this `*.launch` file with a new one created in your own environment that behaves correctly.

**Reply by mmoschetti** [reply]
I already tried to create a new configuration and copy all the settings manually. It worked only the first time I run it, but it failed thereafter. I created a new one again, but this time it never worked.
